planning/route_handler/src/route_handler.cpp

```diff
@@ -715,12 +715,28 @@ bool RouteHandler::getClosestLaneletWithinRoute(
 bool RouteHandler::getNextLaneletWithinRoute(
   const lanelet::ConstLanelet & lanelet, lanelet::ConstLanelet * next_lanelet) const
 {
+  lanelet::ConstLanelet start_lanelet;
+  lanelet::ConstLanelet goal_lanelet;
+  bool flag_check = true;
+  if (
+    !getGoalLanelet(&goal_lanelet) ||
+    !getClosestLaneletWithinRoute(route_msg_.start_pose, &start_lanelet)) {
+    flag_check = false;
+  }
+  if (goal_lanelet.id() == start_lanelet.id()) {
+    flag_check = false;
+  }
```

Comment:
```
  if (exists(goal_lanelets_, lanelet)) {
    return false;
  }
```
doesn't depend on `flag_check` or `start_lanelet` or `goal_lanelet`, so it might be better for this snippet start first (line 718) then followed by lanelet::ConstLanelet start_lanelet etc.

Reply:
@shulanbushangshu 
It looks like the current setup checks for this condition first, before checking the flag.

To make sure things are efficient, I think it would be a good idea to start by performing this check first, before doing anything related to the flag_check. What do you think about this suggestion?